tasmota/xsns_67_as3935.ino

```diff
@@ -150,75 +169,83 @@ void AS3935WriteRegister(uint8_t reg, uint8_t mask, uint8_t shift, uint8_t data)
   data |= currentReg;
   I2cWrite8(AS3935_ADDR, reg, data);
   if (reg == 0x08) Settings.as3935_sensor_cfg[4] = I2cRead8(AS3935_ADDR, reg);
-  if (reg < 0x04) Settings.as3935_sensor_cfg[reg] = I2cRead8(AS3935_ADDR, reg);
+  if (reg <= 0x03) Settings.as3935_sensor_cfg[reg] = I2cRead8(AS3935_ADDR, reg);
 }
 
 /********************************************************************************************/
 // Autotune Caps
-void ICACHE_RAM_ATTR AS3935CountFreq() {
+void ICACHE_RAM_ATTR AS3935CountFreq(void) {
   if (as3935_sensor.dispLCO)
     as3935_sensor.pulse++;
 }
 
 bool AS3935AutoTuneCaps(uint8_t irqpin) {
   int32_t maxtune = 17500; // there max 3.5 % tol
   uint8_t besttune;
+  uint8_t oldvalue = AS3935GetTuneCaps();
   AS3935WriteRegister(LCO_FDIV, 0); // Fdiv 16
   delay(2);
   for (uint8_t tune = 0; tune < 16; tune++) {
     AS3935WriteRegister(TUNE_CAPS, tune);
     delay(2);
-    AS3935WriteRegister(DISP_LCO,1);
+    AS3935WriteRegister(DISP_LCO, 1);
     delay(1);
     as3935_sensor.dispLCO = true;
     as3935_sensor.pulse = 0;
     attachInterrupt(digitalPinToInterrupt(irqpin), AS3935CountFreq, RISING);
-    delay(200); // 100ms callback not work accurat for fequ. measure
+    delay(50);
```

Comment:
I noticed this issue when using the DFRobot AS3935 board. The manufacturer has published the cap value but auto-tune always found the max allowed value. Thanks for addressing this. Will test with this code when I get the change (possibly in a few weeks time).

Reply:
It might make sense to split the AS3935 changes into a separate PR

Reply:
this was not an issue. Should also work with 200ms. Has only been changed because it works exactly the same with this setting.
Please Wait for tests and then i make a new PR for AS3935.

please make new Posts in
https://github.com/arendst/Tasmota/issues/9016

thanks.